### ISSUE TITLE
[TypeScript] go to behaviour is inconsistent

### ISSUE THREAD

**Issue by orangecoloured** [sublimehq/Packages#2717]
- Sublime Version: 4098
- OS Version: macOS Big Sur 11.2.1

## Expected behavior
Whether it is an exported interface, variable, function or class the `go to definition` should behave the same.

## Actual behavior
Two different exported arrow functions demonstrate different behaviour.
I can go to type, but can't go to interface.

## Steps to reproduce

1. https://user-images.githubusercontent.com/3314891/107855926-2009db80-6e2e-11eb-9d6f-f240aeb1b278.mov
2. https://user-images.githubusercontent.com/3314891/107855939-2c8e3400-6e2e-11eb-943b-8e02f8438b90.mov

**Comment by wbond** [comment]
We don’t generally index variables, as that would lead to the symbol index being far less useful in most projects.

However, we should be indexing interfaces.

**Comment by orangecoloured** [comment]
What does it mean, generally?
Why one of the exported functions is accessible and the other is not?
Can there be a preference toggle to force indexing of exported variables?

**Comment by rchl** [comment]
![Screenshot 2021-02-13 at 20 22 04](https://user-images.githubusercontent.com/153197/107859347-2fdaed00-6e39-11eb-9679-36546f193ef2.png)

The first variable is assigned an arrow function which makes the variable be a reference to a function and functions are indexed.

The second variable is just assigned a result of a function call so it's not a function and thus not indedex.

The other issue with `interface` seems valid.

**Comment by rchl** [comment]
BTW. Not to hijack the thread but if you want a real code intelligence that actually understand everything about the code then you'd need something like `LSP` + `LSP-typescript` packages. Sublime Text provides "go to definition" solely based on syntax and that has its limitations.

**Comment by rwols** [comment]
Note that your first video is using a syntax that's not a default syntax.

**Comment by Thom1729** [comment]
PR #2718 adds interface names to the symbol list in TypeScript.

I've also opened an RFC to suggest that interface names (and some other `entity.name` scopes) should be included in the symbol list by default: #2719.

> Can there be a preference toggle to force indexing of exported variables?

By default, we don't add constant names to the symbol list, but I can see how it would be very useful to add the names of exported constants. This would probably require a slight modification to the syntax definition to ensure that we can do this reliably. I've opened issue #2720 if you'd like to add your thoughts.
